src/push.js

```diff
@@ -35,26 +35,31 @@ function ub64(buffer) {
 
 /**
  * Sends a message using the Web Push protocol
- * @param  {String}   message      The message to send
- * @param  {Object}   subscription The subscription details for the client we
- *                                 are sending to
- * @param  {String}   authToken    Optional token to be used in the
- *                                 `Authentication` header if the endpoint
- *                                 requires it.
+ * @param  {String}   message       The message to send
+ * @param  {Object}   subscription  The subscription details for the client we
+ *                                  are sending to
+ * @param  {String}   authToken     Optional token to be used in the
+ *                                  `Authentication` header if the endpoint
+ *                                  requires it.
+ * @param {Number}    paddingLength The number of bytes of padding to add to the
+ *                                  message before encrypting it.
  * @return {Promise} A promise that resolves if the push was sent successfully
  *                   with status and body.
  */
-function sendWebPush(message, subscription, authToken) {
+function sendWebPush(message, subscription, authToken, paddingLength) {
```

Comment:
Could you switch around the paddingLength with the authToken?

I'm still pro just making a special method of GCM Auth header, simplifying sendWebPush method.